hm13_14/ht_template.ipynb

```diff
@@ -0,0 +1,466 @@
+      "|         STOCK GLASS|\n",
+      "+--------------------+\n",
+      "only showing top 20 rows\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# тут має бути розвʼязок задачі\n",
+    "join3 = film_df.join(inventory_df, inventory_df.film_id==film_df.film_id, 'left_outer')\n",
+    "\n",
+    "join3.select('title').distinct().show(20)"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "5.\n",
+    "Вивести топ 3 актори, які найбільше зʼявлялись в категорії фільмів “Children”"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 47,
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "+----------+---------+--------------+\n",
+      "|first_name|last_name|count(film_id)|\n",
+      "+----------+---------+--------------+\n",
+      "|     HELEN|   VOIGHT|             7|\n",
+      "|     SUSAN|    DAVIS|             6|\n",
+      "|     KEVIN|  GARLAND|             5|\n",
+      "+----------+---------+--------------+\n",
+      "only showing top 3 rows\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# тут має бути розвʼязок задачі\n",
+    "children_ct_df = category_df.select('category_id').where(category_df.name == 'Children')\n",
+    "join5 = children_ct_df.join(film_category_df, film_category_df.category_id==children_ct_df.category_id, 'inner').join(film_actor_df, film_actor_df.film_id==film_category_df.film_id, 'inner').join(actor_df, actor_df.actor_id==film_actor_df.actor_id, 'inner')\n",
+    "\n",
+    "result1 = join5.groupBy(actor_df['first_name'], actor_df['last_name']).agg(\n",
+    "    f.count(film_category_df['film_id'])\n",
+    ").orderBy(f.desc(f.count(film_category_df['film_id']))).show(3)\n"
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "markdown",
+   "source": [
+    "6.\n",
+    "Вивести міста з кількістю активних та неактивних клієнтів\n",
+    "(в активних customer.active = 1).\n",
+    "Результат відсортувати за кількістю неактивних клієнтів за спаданням."
+   ],
+   "metadata": {
+    "collapsed": false
+   }
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "outputs": [],
+   "source": [
+    "# тут має бути розвʼязок задачі\n",
+    "join6 = customer_df.join(address_df, address_df.address_id==customer_df.address_id, 'inner').join(city_df, city_df.city_id==address_df.city_id, 'inner')\n",
+    "\n",
+    "#df3 = customer_df.withColumn(\"inactive_customers\", expr(\"case when active = 0 then 1 \" + \"else 0 end\"))\n",
+    "\n",
+    "#df4 = customer_df.select(col(\"*\"), expr(\"case when active == 0 then 1 \" + \"else 0 end\").alias(\"inactive_customers\"))\n",
+    "#df2 = customer_df.withColumn(\"inactive_customers\", when(col(\"active\") == 0, 1).otherwise(0))\n",
+    "#customer_df.select()\n",
+    "\n",
+    "def inact(x: int) -> int:\n",
```

Comment:
It looks like this function is not needed. We have `customer.active = 1` in the data base.